webapp/src/index.tsx

```diff
@@ -295,22 +296,82 @@ export default class Plugin {
             return {message, args};
         });
 
-        const joinCall = (channelID: string, teamID: string) => {
+        const connectToCall = async (channelId: string, teamId: string, title?: string) => {
+            try {
+                const users = voiceConnectedUsers(store.getState());
+                if (users && users.length > 0) {
+                    store.dispatch({
+                        type: VOICE_CHANNEL_PROFILES_CONNECTED,
+                        data: {
+                            profiles: await getProfilesByIds(store.getState(), users),
+                            channelId,
+                        },
+                    });
+                }
+            } catch (err) {
+                logErr(err);
+            }
+
             if (!connectedChannelID(store.getState())) {
-                connectCall(channelID);
+                connectCall(channelId, title);
 
                 // following the thread only on join. On call start
                 // this is done in the call_start ws event handler.
-                if (voiceConnectedUsersInChannel(store.getState(), channelID).length > 0) {
-                    followThread(store, channelID, teamID);
+                if (voiceConnectedUsersInChannel(store.getState(), channelId).length > 0) {
+                    followThread(store, channelId, teamId);
                 }
-            } else if (connectedChannelID(store.getState()) !== channelID) {
+            } else if (connectedChannelID(store.getState()) !== channelId) {
                 store.dispatch({
                     type: SHOW_SWITCH_CALL_MODAL,
                 });
             }
         };
 
+        const joinCall = async (channelId: string, teamId: string, title?: string): Promise<SlashCommandWillBePostedReturn> => {
```

Comment:
I am not against explicit returns but what I find odd here is that we are coupling this generic join function with a special case (slash command). If you'd like to keep the return then maybe we should consider returning an error or null instead.

Reply:
@streamer45 Ah, I think I understand what you mean now -- since joinCall is common, don't specialize its return to the slash command usecase. Gotcha. In that case throwing seems right, thanks!

Btw, in testing I found something interesting -- returned messages from slash commands (from the webapp) don't appear in the RHS. Only when you're in the centre channel.  That's a shame.  Hm, this might be an edge case for the original willBePosted implementation...

Reply:
Yeah, not something we can fix here I am afraid.